driver/src/main/java/org/neo4j/driver/StatementResult.java

```diff
@@ -138,29 +138,12 @@
      */
     <T> List<T> list( Function<Record, T> mapFunction );
 
-    /**
-     * Consume the entire result, yielding a summary of it.
-     *
-     * Calling this method exhausts the result.
-     *
-     * <pre class="doctest:ResultDocIT#summarizeUsage">
-     * {@code
-     * ResultSummary summary = session.run( "PROFILE MATCH (n:User {id: 12345}) RETURN n" ).consume();
-     * }
-     * </pre>
-     *
-     * @return a summary for the whole query result
-     */
-    ResultSummary consume();
-
     /**
      * Return the result summary.
      *
-     * If the records in the result is not fully consumed, then calling this method will force to pull all remaining
-     * records into buffer to yield the summary.
+     * If the records in the result is not fully consumed, then calling this method will exhausts the result.
```

Comment:
Maybe clarify that exhausting means exhausting the current batch, not the whole possible result set / size.

Reply:
No, it will exhausting the whole result. It is equivalent to `Subscription#cancel`.

We currently cannot discard N, we can only discard ALL.

The following code is equivalent to a driver user:
```
// Code with 1.7 driver
public ResultSummary run() {
	try (AutoCloseableStatementRunner statementRunner = getStatementRunner(this.targetDatabase)) {
		StatementResult result = runnableStatement.runWith(statementRunner);
		return result.consume();
	}
}

// Code with 4.0 driver
public ResultSummary run() {
	try (AutoCloseableStatementRunner statementRunner = getStatementRunner(this.targetDatabase)) {
		StatementResult result = runnableStatement.runWith(statementRunner);
		return result.summary();
	}
}
```